File: lectures/lecture12/firstomp.f90
!Getting started with OpenMP
!To compile: gfortran -fopenmp -o first.exe firstomp.f90
!To run: $ ./first.exe
program firstomp
	use omp_lib !makes OpenMP routines, variables available
	implicit none
	integer :: NumThreads,threadID

!$OMP PARALLEL PRIVATE(threadID)
	NumThreads = omp_get_num_threads()
	threadID = omp_get_thread_num()
	print *, 'this is thread',threadID, ' of ', NumThreads
!$OMP END PARALLEL

    print *, 'exited parallel region, this is thread', threadID

end program firstomp

File: lectures/lecture12/firstomp.cpp
//Getting started with OpenMP
#include <iostream>
#include <omp.h>
int main()
{
#pragma omp parallel
	{
	int NumThreads,threadID;
    NumThreads = omp_get_num_threads();
	threadID = omp_get_thread_num();
#pragma omp critical
{
    std::cout << "This is thread " << threadID << std::endl;
	}
    }

return 0;
}

